components/IndicatorStat/utils/data.ts

```diff
@@ -176,3 +176,11 @@ export const getIndicatorsValue = (
 
   return 'Invalid Configuration';
 };
+
+export const setDecimalCount = (flag: string | undefined): number => {
+  if (flag) {
+    return flag.indexOf('th') === 0 ? 0 : 1;
+  }
+
+  return 1;
```

Comment:
I've just noticed that you're using this to set the decimal count, but it doesn't take in the actual decimalCount as configured in the valueOptions (or with stat or chart). The default should be whatever is initially configured as the decimalCount.

Reply:
Now passing default decimalCount value from `IndicatorStat` to `setDecimalCount` function